rmf_battery/include/rmf_battery/agv/MechanicalSystem.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_BATTERY__AGV__MECHANICALSYSTEM_HPP
+#define RMF_BATTERY__AGV__MECHANICALSYSTEM_HPP
+
+#include <rmf_utils/impl_ptr.hpp>
+
+namespace rmf_battery {
+namespace agv {
+
+
+class MechanicalSystem
+{
+public:
+
+  MechanicalSystem(
+    double mass,
+    double inertia,
+    double friction_coefficient);
+
+  MechanicalSystem& mass(double mass);
+  double mass() const;
+
+  MechanicalSystem& inertia(double inertia);
+  double inertia() const;
+
+  MechanicalSystem& friction_coefficient(double friction_coeff);
+  double friction_coefficient() const;
```

Comment:
What type of friction is this meant to be? Static friction? Kinetic friction?

What are we dragging that's creating friction? Is this meant to be friction in the rotation of a vehicle's wheels? Or just the general forces on the mechanical system that push against its motion?

For a user to know what kind of friction coefficient they provide, we should be explicit about what calculations we will use it for.

Reply:
This is the kinetic friction in the rotation of the vehicles's wheels which is used to calculate the energy loss during locomotion. I will update the documentation to reflect this.

Reply:
https://github.com/osrf/rmf_core/pull/217/commits/d416f83a321ccf04a08fd1711248194babee5fbc